dev/io.openliberty.org.apache.myfaces.3.0/src/org/apache/myfaces/config/FacesConfigurator.java

```diff
@@ -503,100 +481,61 @@ public void update()
-        {
-            lifecycleFactory = lifecycleFactory.getWrapped(); // MYFACES-4469
-        }
-        //lifecyclePurgeMethod = lifecycleFactory.getClass().getMethod("purgeLifecycle", NO_PARAMETER_TYPES);
-        lifecyclePurgeMethod = getPurgeMethod(lifecycleFactory, "purgeLifecycle", NO_PARAMETER_TYPES);
+        purge(lifecycleFactory);
 
         FacesContext facesContext = getFacesContext();
-        while(!(facesContext instanceof org.apache.myfaces.context.servlet.FacesContextImpl)) {
-            if(facesContext instanceof jakarta.faces.context.FacesContextWrapper)
-            facesContext = ((jakarta.faces.context.FacesContextWrapper)facesContext).getWrapped(); // MYFACES-4469
-        }
-        facesContextPurgeMethod = getPurgeMethod(facesContext, "purgeFacesContext", NO_PARAMETER_TYPES);
-        
-        // If there was no exception so far, now we can purge
-        //
-        if (appFactoryPurgeMethod != null && renderKitPurgeMethod != null && lifecyclePurgeMethod != null && 
-            facesContextPurgeMethod != null)
-        {
-            appFactoryPurgeMethod.invoke(applicationFactory, NO_PARAMETERS);
-            renderKitPurgeMethod.invoke(renderKitFactory, NO_PARAMETERS);
-            RuntimeConfig.getCurrentInstance(_externalContext).purge();
-            lifecyclePurgeMethod.invoke(lifecycleFactory, NO_PARAMETERS);
-            facesContextPurgeMethod.invoke(facesContext, NO_PARAMETERS);
-
-            // factories and serial factory need not be purged...
+        purge(facesContext);
 
-            // Remove first request processed so we can initialize it again
-            _externalContext.getApplicationMap().remove(LifecycleImpl.FIRST_REQUEST_PROCESSED_PARAM);
-            return true;
-        }
-        return false;
+        _externalContext.getApplicationMap().remove(LifecycleImpl.FIRST_REQUEST_PROCESSED_PARAM);
     }
     
-    private Method getPurgeMethod(Object instance, String methodName, Class<?>[] parameters)
-    {
+    <T> void purge(T instance)
+    {        
         while (instance != null)
         {
-            Method purgeMethod = null;
-            try
+            if (instance instanceof Purgeable)
             {
-                purgeMethod = instance.getClass().getMethod(methodName, parameters);
+                ((Purgeable) instance).purge();
             }
-            catch (NoSuchMethodException e)
-            {
-                // No op, it is expected to found this case, so in that case
-                // look for the parent to do the purge
-            }
-            if (purgeMethod != null)
+
+            if (instance instanceof FacesWrapper)
             {
-                return purgeMethod;
+                instance = ((FacesWrapper<T>) instance).getWrapped();
             }
-            if (instance instanceof FacesWrapper)
+            else
             {
-                instance = ((FacesWrapper)instance).getWrapped();
+                return;
```

Comment:
Looking at the upstream commit it looks like this was missed and needs to be applied to your 3.0.x PR

Reply:
https://github.com/volosied/myfaces/blob/9ccc1a48bf7a0ba9c46725a1b1708e2992f72957/impl/src/main/java/org/apache/myfaces/config/FacesConfigurator.java#L531

Reply:
Fixed: https://github.com/apache/myfaces/commit/218dd53ce0d38263fb74f6d49b70b5b15671eeeb